torch_dicom/preprocessing/crop.py

```diff
@@ -416,3 +418,105 @@ def get_bounds(self, x: Tensor, sopuid: SOPUID) -> Tensor:
             bounds = torch.tensor([x1, y1, x2, y2])
 
         return bounds
+
+
+@dataclass
+class TileCrop(Crop):
+    r"""Performs cropping of an image into tiles.
+
+    Note:
+        Because this transform creates multiple tiles, it should not be followed by other image transforms.
+        Additionally, :func:`apply_to_coords` and :func:`unapply_to_coords` will not work.
+
+    Args:
+        img_key: Key for the image in the example dict.
+
+        bounds_key: Key for the crop bounds in the example dict. The determined
+            crop bounds will be stored in this location.
+
+        img_dest_key: Key for the cropped image in the example dict. If None,
+            the cropped image will be stored in the same location as the original image.
+
+        size: Size of the tiles in the format (H, W).
+
+        overlap: Amount of overlap between tiles (in fractional units) in the format (H, W).
+
+    Shape:
+        - Input: :math:`(N, *, H, W)`
+        - Output: :math:`(N, *, D, H', W')`
+    """
+    size: Tuple[int, int] = (256, 256)
+    overlap: Tuple[float, float] = (0.2, 0.2)
+
+    def __call__(self, example: E) -> E:
+        # Get the tile bounds
+        img: Tensor = example["img"]
+        bounds = self.get_bounds(img)
+
+        # TODO: Consider adding crop metadata and fixing apply_to_coords / unapply_to_coords
+        # if it becomes necessary to invert the crops
+        # Apply each crop
+        images: List[Tensor] = [super(TileCrop, self).__call__(copy(example), b)["img"] for b in bounds]
+
+        # Stack into a new "depth" dimension
+        example["img"] = torch.stack(images, dim=-3)
+
+        return example
+
+    @torch.no_grad()
+    def get_bounds(self, x: Tensor) -> Tensor:
+        r"""Returns the crop bounds for the given image. Bounds are determined by
+        splitting the image into tiles of size `self.size` with overlap `self.overlap`.
+
+        Args:
+            x: Image tensor.
+
+        Shape:
+            - Input: :math:`(*, H, W)`
+            - Output: :math:`(N, 4)` where :math:`N` is the number of tiles.
+
+        Returns:
+            Absolute crop bounds in the format :math:`(x_1, y_1, x_2, y_2)`.
+        """
+        if x.ndim < 2:
+            raise ValueError(f"Expected input to have at least 2 dimensions, got {x.ndim}")
+        H, W = x.shape[-2:]
+        Hc, Wc = self.size
+
+        # Determine overlap in pixels
+        overlap_h = int(self.overlap[0] * Hc)
+        overlap_w = int(self.overlap[1] * Wc)
+
+        # Calculate the number of tiles in each dimension
+        num_tiles_h = math.ceil(H / (Hc - overlap_h))
+        num_tiles_w = math.ceil(W / (Wc - overlap_w))
+
+        # Calculate the starting points for each tile
+        start_h = (torch.arange(num_tiles_h) * (Hc - overlap_h)).clip(max=H - Hc)
```

Comment:
Not necessary, but it could be interesting at some point to dynamically calculate overlap: i.e., overlap `>= {self.overlap}` is calculated such that overlap is uniform across all tiles (instead of clipping).